mapusaurus/hmda/tests/test_views.py

```diff
@@ -40,18 +43,74 @@ def mkrecord(action_taken, countyfp, geoid):
                 ffieic_median_family_income='1000', tract_to_msamd_income='1000',
                 number_of_owner_occupied_units='1', number_of_1_to_4_family_units='1',
                 application_date_indicator=1)
-        
-            record.geo_geoid = geoid
-            record.institution_id = respondent.institution_id
+            record.geo = geo
+            record.institution = respondent
             record.save()
 
-        mkrecord(1, '222', '1122233300')
-        mkrecord(1, '222', '1122233300')
-        mkrecord(1, '222', '1122233400')
-        mkrecord(8, '222', '1122233300')
-        mkrecord(1, '222', '1122233300')
-        mkrecord(1, '223', '1122333300')
+        #institution #1 records, total = 6; selected institution
+        mkrecord("91000000001", 1, '222', '1122233300')
+        mkrecord("91000000001", 1, '222', '1122233300')
+        mkrecord("91000000001", 1, '222', '1122233400')
+        mkrecord("91000000001", 1, '222', '1122233300')
+        mkrecord("91000000001", 1, '222', '1122233300')
+        mkrecord("91000000001", 1, '223', '1122333300')
+
+        #institution #2 records, total = 4; >.5 and <2.0; peer
+        mkrecord("91000000002", 1, '222', '1122233300')
+        mkrecord("91000000002", 1, '222', '1122233300')
+        mkrecord("91000000002", 1, '222', '1122233400')
+        mkrecord("91000000002", 1, '222', '1122233300')
+        
+        #institution #3 records, total =2; <.5; not peer
+        mkrecord("91000000003", 1, '222', '1122233300')
+        mkrecord("91000000003", 1, '223', '1122333300')
+
+        #institution #4 records, total = 13; >2.0; not peer
+        mkrecord("11000000001", 1, '222', '1122233300')
+        mkrecord("11000000001", 1, '222', '1122233300')
+        mkrecord("11000000001", 1, '222', '1122233400')
+        mkrecord("11000000001", 1, '222', '1122233300')
+        mkrecord("11000000001", 1, '222', '1122233300')
+        mkrecord("11000000001", 1, '223', '1122333300')
+        mkrecord("11000000001", 1, '222', '1122233300')
+        mkrecord("11000000001", 1, '222', '1122233300')
+        mkrecord("11000000001", 1, '222', '1122233400')
+        mkrecord("11000000001", 1, '222', '1122233300')
+        mkrecord("11000000001", 1, '222', '1122233300')
+        mkrecord("11000000001", 1, '223', '1122333300')
+        mkrecord("11000000001", 1, '223', '1122333300')
+
+        #institution #4 records, total = 4; >.5 and <2.0; peer
+        mkrecord("11000000002", 1, '222', '1122233300')
+        mkrecord("11000000002", 1, '222', '1122233300')
+        mkrecord("11000000002", 1, '222', '1122233400')
+        mkrecord("11000000002", 1, '222', '1122233300')
+        mkrecord("11000000002", 1, '222', '1122233300')
+        call_command('calculate_loan_stats')
```

Comment:
This is fine, but I think there's a lesson to be learned here.  In general, you shouldn't have a lot of business logic in your management commands, as it gets messy if you ever need to use that logic in your code (as in this case).  Typically I'd put the logic in models.py or by creating a new class in base namespace, and the management command would simply be a wrapper to execute it.  I think it's a lot of work to refactor just for this test, but I figured I'd point it out anyway.